modules/lang-painless/src/main/java/org/elasticsearch/painless/action/PainlessExecuteAction.java

```diff
@@ -647,9 +649,9 @@ private static Response prepareRamIndex(Request request,
                     BytesReference document = request.contextSetup.document;
                     XContentType xContentType = request.contextSetup.xContentType;
                     SourceToParse sourceToParse = new SourceToParse(index, "_id", document, xContentType);
-                    //TODO this throws NPE when called against an empty index with no provided mappings: DocumentMapper is null
-                    // and the corresponding empty MappingLookup does not have a DocumentParser set
-                    ParsedDocument parsedDocument = indexService.mapperService().mappingLookup().parseDocument(sourceToParse);
+                    MappingLookup mappingLookup = indexService.mapperService().mappingLookup();
```

Comment:
this is much better than before, as you don't get a null pointer, and mapping lookup is not responsible for parsing documents, but in case an index has no mappings, the mapping lookup won't hold metadata fields. Maybe it would be more correct in this special case to allow to create a new mapping lookup that is parse proof, but I am not entirely sure that this is a big problem.

Reply:
On the long run, if we move to eagerly creating mappings for any new index, this problem goes away entirely so maybe it is not worth investing more effort on it.

Reply:
For the record, I decided to leave things as they are and leave a comment to explain the situation.

1.